libdnf/sack/advisorycollection.cpp

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2020 Red Hat, Inc.
+ *
+ * Licensed under the GNU Lesser General Public License Version 2.1
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#include "advisorymodule.hpp"
+#include "../dnf-sack-private.hpp"
+#include "advisorycollection.hpp"
+
+namespace libdnf {
+
+class AdvisoryCollection::Impl {
+private:
+    friend AdvisoryCollection;
+    DnfSack *sack;
+    Id advisory;
+    std::vector<AdvisoryPkg> pkgList;
+    std::vector<AdvisoryModule> moduleList;
+};
+
+AdvisoryCollection::AdvisoryCollection(DnfSack *sack, Id advisory, std::vector<AdvisoryPkg> pkgList, std::vector<AdvisoryModule> moduleList)
+: pImpl(new Impl)
+{
+    pImpl->sack = sack;
+    pImpl->advisory = advisory;
+    pImpl->pkgList = pkgList;
+    pImpl->moduleList = moduleList;
+}
+
+AdvisoryCollection::AdvisoryCollection(const AdvisoryCollection & src) : pImpl(new Impl) { *pImpl = *src.pImpl; }
+AdvisoryCollection::AdvisoryCollection(AdvisoryCollection && src) : pImpl(new Impl) { pImpl.swap(src.pImpl); }
+AdvisoryCollection::~AdvisoryCollection() = default;
+
+AdvisoryCollection & AdvisoryCollection::operator=(const AdvisoryCollection & src) { *pImpl = *src.pImpl; return *this; }
+
+AdvisoryCollection &
+AdvisoryCollection::operator=(AdvisoryCollection && src) noexcept
+{
+    pImpl.swap(src.pImpl);
+    return *this;
+}
+
+bool AdvisoryCollection::isApplicable() const {
+    auto moduleContainer = dnf_sack_get_module_container(pImpl->sack);
+    if (!moduleContainer) {
+        return true;
+    }
+
+     for (auto & moduleAdvisory: pImpl->moduleList) {
```

Comment:
I see 5 spaces here and not 4. Please could you check it?

Reply:
It was removed.